CommonFindPackage.cmake

```diff
@@ -269,4 +269,8 @@ macro(common_find_package_post)
   endif()
   common_graph(${PROJECT_NAME})
   message(STATUS ${__configure_msg})
+
+  # If this variable was given in the command line, ensure that the package
+  # installation is only run in this cmake invocation.
+  unset(INSTALL_PACKAGES CACHE)
```

Comment:
Limiting interactions across the different files really helps with modularity and maintainability.
After looking a bit I think that it can in fact be moved to the end of SubProject.cmake. We could make a check for ${name}_IS_SUBPROJECT to guarantee that it is only executed by the top-level project.